#! /usr/bin/env python

PACKAGE='aricc_2d_vision'
try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("width_tolerance",  double_t, 0, "The width tolerance", 0.1, 0.0, 1.0)
gen.add("height_tolerance",  double_t, 0, "The height tolerance", 0.1, 0.0, 1.0)
gen.add("color_h_tol", double_t, 0, "The color h tolerance",    1.0, 0.0, 255.0)
gen.add("color_s_tol", double_t, 0, "The color s tolerance",    1.0, 0.0, 255.0)
gen.add("color_v_tol", double_t, 0, "The color v tolerance",    1.0, 0.0, 255.0)
gen.add("density_tolerance", double_t, 0, "The density tolerance",    1.0, 0.0, 255.0)
gen.add("debug",      bool_t, 0, "the switch of debug", False)
gen.add("print_info", bool_t, 0, "the switch of printing info", False)

exit(gen.generate(PACKAGE, "aricc_2d_vision", "RectBasedObjectDetection"))
